src/ompl/tools/thunder/Thunder.h

```diff
@@ -0,0 +1,210 @@
+
+            /** \brief Set the planner allocator to use. This is only
+                used if no planner has been set. This is optional -- a default
+                planner will be used if no planner is otherwise specified. */
+            void setPlannerAllocator(const base::PlannerAllocator &pa);
+
+            /** \brief Run the planner for up to a specified amount of time (default is 1 second) */
+            virtual base::PlannerStatus solve(double time = 1.0);
+
+            /** \brief Run the planner until \e ptc becomes true (at most) */
+            virtual base::PlannerStatus solve(const base::PlannerTerminationCondition &ptc);
+
+            /** \brief Save the experience database to file */
+            bool save();
+
+            /** \brief Save the experience database to file if there has been a change */
+            bool saveIfChanged();
+
+            /** \brief Clear all planning data. This only includes
+                data generated by motion plan computation. Planner
+                settings, start & goal states are not affected. */
+            virtual void clear(void);
+
+            /** \brief Print information about the current setup */
+            virtual void print(std::ostream &out = std::cout) const;
+
+            /** \brief This method will create the necessary classes
+                for planning. The solve() method will call this function automatically. */                
+            virtual void setup(void);
+
+            /** \brief Get a vector of all the planning data in the database */
+            void getAllPlannerDatas(std::vector<ompl::base::PlannerDataPtr> &plannerDatas) const;
+
+            /** \brief Get the total number of paths stored in the database */
+            std::size_t getExperiencesCount() const;
+
+            /** \brief Convert PlannerData to PathGeometric. Assume ordering of verticies is order of path */
+            void convertPlannerData(const ompl::base::PlannerDataPtr plannerData, ompl::geometric::PathGeometric &path);
+
+            /**
+             * \brief If path1 and path2 have a better start/goal match when reverse, then reverse path2
+             * \param path to test against
+             * \param path to reverse
+             * \return true if reverse was necessary
+             */
+            bool reversePathIfNecessary(ompl::geometric::PathGeometric &path1, ompl::geometric::PathGeometric &path2);
+
+            /** \brief Hook for getting access to debug data */
+            ompl::tools::ThunderDBPtr getExperienceDB();
+
+            /** \brief Allow accumlated experiences to be processed */             
+            bool doPostProcessing();
+
+        protected:
+
+            /**  The maintained experience planner instance */
+            base::PlannerPtr                             rrPlanner_;
+
+            /**  A third planner used for testing dual-thread scratch-only planning */
+            base::PlannerPtr                             planner2_;
```

Comment:
No, except that I have my own experience-planning benchmarking functionality that no other planners in OMPL need (located in ExperienceSetup.h), and by having the dual planning functionality here it makes benchmarking much much easier